qui/tray/gross_gtk3_bug_workaround.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3 --
+import os
+import sys
+from typing import Optional
+
+# Modifying the environment while multiple threads
+# are running leads to use-after-free in glibc, so
+# ensure that only one thread is running.
+if len(os.listdir("/proc/self/task")) != 1:
```

Comment:
in what situation can this happen? when? what should the user do? If this is a real thing that can happen, this needs an error that helps user fix the situation, if it's not, it shouldn't be here.

Reply:
IIUC this happens only when the module is imported too late. So, it's an assert for a programmer error, not a runtime condition. But then, if you make such a mistake, debugging it may be quite tricky (see all the cases of debugging SEGV somewhere in GTK etc), so this assert will help to find it early.

Reply:
@marmarek you understand correctly.